### PR TITLE
Multiple adjustments for Xaml Basics

### PR DESCRIPTION
**Pull request by jeromelaban**
## PR Type
- Bugfix
- Feature

## What is the new behavior?
- Move `VisibleBoundsPadding` inside Uno.UI so Default Styles can use it
- Fix for `NavigationView` toggle button initial location on iOS
- Close the `NavigationView` pane when an item is selected
- Build Uno.UI.Toolkit for Xamarin.Mac

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [x] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [x] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)

### CODE REVIEW

File: src/Uno.UI.Toolkit/Uno.UI.Toolkit.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0">
 	<PropertyGroup>
-		<TargetFrameworks>MonoAndroid80;uap10.0;xamarinios10;netstandard2.0</TargetFrameworks>
-		<TargetFrameworksCI>MonoAndroid71;MonoAndroid80;uap10.0;xamarinios10;netstandard2.0</TargetFrameworksCI>
+		<TargetFrameworks>xamarinmac20;MonoAndroid80;uap10.0;xamarinios10;netstandard2.0</TargetFrameworks>
+		<TargetFrameworksCI>MonoAndroid71;MonoAndroid80;uap10.0;xamarinios10;netstandard2.0mxamarinmac20</TargetFrameworksCI>
```
Comment (davidjohnoliver) [inline src/Uno.UI.Toolkit/Uno.UI.Toolkit.csproj]: missing semicolon

File: src/Uno.UI/UI/Xaml/Controls/NavigationView/NavigationView.cs [hunk 2]
```diff
@@ -83,7 +83,7 @@ private void UpdateMinimalMode()
 
 			if (_togglePaneButton != null && _navigationViewBackButton != null)
 			{
-				_togglePaneButton.Margin = new Thickness(0, _navigationViewBackButton.RenderSize.Height, 0, 0);
+				_togglePaneButton.Margin = new Thickness(0, _navigationViewBackButton.Height, 0, 0);
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/NavigationView/NavigationView.cs]: why this change? What's going to happen if Height isn't set? I don't even know what a Thickness with NaN in it does.

File: src/Uno.UI/UI/Xaml/Style/Generic/Generic.xaml [hunk 3]
```diff
@@ -9652,7 +9653,8 @@
                             PaneBackground="{ThemeResource NavigationViewDefaultPaneBackground}">
 
                             <SplitView.Pane>
-                                <Grid x:Name="PaneContentGrid">
+                                <Grid x:Name="PaneContentGrid"
+										uBehaviors:InternalVisibleBoundsPadding.PaddingMask="All">
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Style/Generic/Generic.xaml]: Maybe a comment here that we're modifying the UWP default template, and why

File: src/Uno.UI/UI/Xaml/Controls/NavigationView/NavigationView.cs (hunk 2 above)
Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/NavigationView/NavigationView.cs]: RenderSize is not updated properly when calling this method, We could problably fail if the size if not properly set, as a last resource. This should have been done as a visual state, but that's not how the default style works...
